contracts/poolMaster/PoolMaster.sol

```diff
@@ -0,0 +1,298 @@
+  IERC20Ext public newKnc;
+  IERC20Ext private oldKnc;
+
+  modifier onlyAdminOrOperator() {
+    require(msg.sender == admin || isOperator(msg.sender), 'only admin or operator');
+    _;
+  }
+
+  receive() external payable {}
+
+  constructor(
+    string memory _name,
+    string memory _symbol,
+    IKyberNetworkProxy _kyberProxy,
+    IKyberStaking _kyberStaking,
+    IKyberGovernance _kyberGovernance,
+    IRewardsDistributor _rewardsDistributor,
+    uint256 _adminFeeBps
+  ) ERC20(_name, _symbol) PermissionAdmin(msg.sender) {
+    kyberProxy = _kyberProxy;
+    kyberStaking = _kyberStaking;
+    kyberGovernance = _kyberGovernance;
+    rewardsDistributor = _rewardsDistributor;
+    newKnc = IERC20Ext(address(_kyberStaking.kncToken()));
+    oldKnc = IERC20Ext(INewKNC(address(newKnc)).oldKNC());
+    oldKnc.safeApprove(address(newKnc), MAX_UINT);
+    newKnc.safeApprove(address(_kyberStaking), MAX_UINT);
+    _changeAdminFee(_adminFeeBps);
+  }
+
+  function changeKyberProxy(IKyberNetworkProxy _kyberProxy) external onlyAdmin {
+    kyberProxy = _kyberProxy;
+  }
+
+  function changeRewardsDistributor(IRewardsDistributor _rewardsDistributor) external onlyAdmin {
+    rewardsDistributor = _rewardsDistributor;
+  }
+
+  function changeGovernance(IKyberGovernance _kyberGovernance) external onlyAdmin {
+    kyberGovernance = _kyberGovernance;
+  }
+
+  function changeAdminFee(uint256 _adminFeeBps) external onlyAdmin {
+    _changeAdminFee(_adminFeeBps);
+  }
+
+  function addOperator(address newOperator) external onlyAdmin {
+    operators[newOperator] = true;
+  }
+
+  function removeOperator(address newOperator) external onlyAdmin {
+    operators[newOperator] = false;
+  }
+
+  function depositWithOldKnc(uint256 tokenWei) external {
+    oldKnc.safeTransferFrom(msg.sender, address(this), tokenWei);
+    INewKNC(address(newKnc)).mintWithOldKnc(tokenWei);
+    _deposit();
+  }
+
+  function depositWithNewKnc(uint256 tokenWei) external {
+    newKnc.safeTransferFrom(msg.sender, address(this), tokenWei);
+    _deposit();
+  }
+
+  /*
+   * @notice Called by users burning their token
+   * @dev Calculates pro rata KNC and redeems from staking contract
+   * @param tokensToRedeem
+   */
+  function withdraw(uint256 tokensToRedeemTwei) external nonReentrant {
+    require(balanceOf(msg.sender) >= tokensToRedeemTwei, 'insufficient balance');
+
+    uint256 proRataKnc = getLatestStake().mul(tokensToRedeemTwei).div(totalSupply());
+    _unstake(proRataKnc);
+    super._burn(msg.sender, tokensToRedeemTwei);
+
+    newKnc.safeTransfer(msg.sender, proRataKnc);
+  }
+
+  /*
+   * @notice Vote on KyberDAO campaigns
+   * @dev Admin calls with relevant params for each campaign in an epoch
+   * @param proposalId: DAO proposalId
+   * @param optionBitMask: voting option
+   */
+  function vote(uint256 proposalId, uint256 optionBitMask) external onlyAdminOrOperator {
+    kyberGovernance.submitVote(proposalId, optionBitMask);
```

Comment:
we should support vote for multiple proposals at once

Reply:
yas